db/sqlglue.c

```diff
@@ -2809,10 +2849,9 @@ static inline int sqlite3VdbeCompareRecordPacked(KeyInfo *pKeyInfo, int k1len,
         return i64cmp(key1, key2);
     }
 
-    char *pFree = 0;
     UnpackedRecord *rec;
 
-    rec = sqlite3VdbeAllocUnpackedRecord(pKeyInfo, NULL, 0, &pFree);
+    rec = sqlite3VdbeAllocUnpackedRecord(pKeyInfo);
     if (rec == 0) {
         logmsg(LOGMSG_ERROR, "Error rec is zero, returned from "
                         "sqlite3VdbeAllocUnpackedRecord()\n");
```

Comment:
wait .. is this a malloc failure?  should we abort?  Joe points out that instead we're returning "YES THESE ARE EQUAL" ..

Reply:
I agree that this is sub-optimal; however, I think changing this within the scope of this pull request is a bit risky.